### PR TITLE
Fix TS issues and lint the release post tool by running `tsc`

### PR DESCRIPTION
**Pull request by samueljseay**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

If we are to make use of TypeScript properly in tools we need to actually lint it at PR time, since we can't rely on IDEs to catch all issues and right now it leads us to leave a lot of TS errors in the code lying around.

This fixes all the lint issues in release post generator to start with and adds it to the CI lint job so that we can't merge without TS compiling. 

- [x] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by roykho — DISMISSED** [review]
LGTM!

File: .github/workflows/pr-lint-test-js.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
               uses: ./.github/actions/setup-woocommerce-monorepo
 
             - name: Lint
-              run: pnpm run -r --filter='woocommerce/client/admin...' --filter='!@woocommerce/e2e*' --filter='!@woocommerce/api' --color lint
+              run: pnpm run -r --filter='tools/release-posts' --filter='woocommerce/client/admin...' --filter='!@woocommerce/e2e*' --filter='!@woocommerce/api' --color lint
```
Comment (jonathansadowski) [inline .github/workflows/pr-lint-test-js.yml]:
```suggestion
              run: pnpm run -r --filter='release-posts' --filter='woocommerce/client/admin...' --filter='!@woocommerce/e2e*' --filter='!@woocommerce/api' --color lint
```

I'm not sure that the filter you originally suggested actually matches. If I ran it without the others locally, I got an error that there weren't any matches. This updated one matches and runs the lint script that was added to the project.

**Review by jonathansadowski — CHANGES_REQUESTED** [review]
I am not sure the workflow change is filtering correctly

File: .github/workflows/pr-lint-test-js.yml (hunk 1 above)
Comment (samueljseay) [reply .github/workflows/pr-lint-test-js.yml]: Ah wow. thanks for spotting. I'll commit your suggestion and check the CI

**Review by jonathansadowski — APPROVED** [review]
Thanks. After the change, looks good to me 👍   Nice work

**Pull request merged**